### PR TITLE
Fixed a bug. The "apps" key is not present in my messages this caused…

### PR DESCRIPTION
**Pull request by Rehan-Mirza**
… an exception and no event was raised
This will fix bug #758

### CODE REVIEW

File: Tweetinvi.Streams/AccountActivityStream.cs [hunk 1]
```diff
@@ -228,18 +228,20 @@ private void TryRaiseUserEvent(string eventName, JObject jsonObjectEvent)
         private void TryRaiseMessageEvent(string eventName, JObject jsonObjectEvent)
         {
             var messageEventDTOs = jsonObjectEvent[eventName].ToObject<EventDTO[]>();
-            var apps = jsonObjectEvent["apps"].ToObject<Dictionary<string, App>>();
+            var apps = jsonObjectEvent.GetValue("apps")?.ToObject<Dictionary<string, App>>();
```
Comment (linvi) [inline Tweetinvi.Streams/AccountActivityStream.cs]: Quick question, for my knowledge. Is there any case where the "apps" property would not exist on such an object?

File: Tweetinvi.Logic/DTO/MessageUser.cs [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        [JsonProperty("statuses_count")]
+        public int StatusesCount { get; set; }
+
+        [JsonProperty("profile_image_url")]
+        public string ProfileImageUrl { get; set; }
+
+        [JsonProperty("profile_image_url_https")]
+        public string ProfileImageUrlHttps { get; set; }
+
+        [JsonProperty("created_timestamp")]
```
Comment (linvi) [inline Tweetinvi.Logic/DTO/MessageUser.cs]: This is the only field that is different from the `UserDTO` class.
Also this field should be a `DateTime`

``` c#
[JsonProperty("created_timestamp")]
[JsonConverter(typeof(JsonTwitterDateTimeConverter))]
public DateTime CreatedTimestamp { get; set; }
```

File: Tweetinvi.Logic/Message.cs [hunk 3]
```diff
@@ -103,6 +103,8 @@ public IMessageEntities Entities
 
         public IMediaEntity AttachedMedia => EventDTO.MessageCreate.MessageData.Attachment?.Media;
 
+        public IMessageUser[] DirectMessageUsers { get; set; }
```
Comment (linvi) [inline Tweetinvi.Logic/Message.cs]: As per the previous comment, this would become : `IUser[]`

File: Tweetinvi.Streams/AccountActivityStream.cs [hunk 4]
```diff
@@ -228,18 +228,20 @@ private void TryRaiseUserEvent(string eventName, JObject jsonObjectEvent)
...
             {
                 App app = null;
 
                 if (messageEventDTO.MessageCreate.SourceAppId != null)
                 {
-                    apps.TryGetValue(messageEventDTO.MessageCreate.SourceAppId.ToString(), out app);
+                    apps?.TryGetValue(messageEventDTO.MessageCreate.SourceAppId.ToString(), out app);
                 }
 
                 var message = _messageFactory.GenerateMessageFromEventDTO(messageEventDTO, app);
```
Comment (linvi) [inline Tweetinvi.Streams/AccountActivityStream.cs]: As it is a factory, it should take the `users` as a new parameter instead of assigning the value.

**Comment by Rehan-Mirza** [comment]
Hi,

I'm out of town right now and bearly have internet access, I'll get back to you in a couple of days.

Rehan

/Rehan

________________________________

File: Tweetinvi.Logic/DTO/MessageUser.cs (hunk 2 above)
Comment (Rehan-Mirza) [reply Tweetinvi.Logic/DTO/MessageUser.cs]: Hi Linvi,

You can reuse the code and do a different check in, that is ok for me, thanks!

/Rehan

**Comment by linvi** [comment]
This issue was fixed by https://github.com/linvi/tweetinvi/pull/804 and will be available in 5.0

**Pull request closed without merging**